granadilla/views.py

```diff
@@ -211,6 +211,34 @@ def get_context_data(self, **kwargs):
 group = login_required(GroupView.as_view())
 
 
+class ChangePasswordView(FormView):
+    """
+    function to change the user's password
+    """
+    model = models.LdapUser
+
+    template_name = 'granadilla/user_password.html'
+    form_class = LdapUserPassForm
+    success_url = 'confirmation/'
+
+    def get_context_data(self, **kwargs):
+        context = super(ChangePasswordView, self).get_context_data(**kwargs)
+        context['form_user'] = self.get_form()
+        return context
+
+    def form_valid(self, form):
+        form.save()
+        return super(ChangePasswordView, self).form_valid(form)
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs['user'] = self.model.objects.get(username=self.request.user.username)
```

Comment:
This could be shortened to ``kwargs['user'] = self.request.user``, and you can then remove the ``model`` declaration.